luigi/server.py

```diff
@@ -168,10 +169,11 @@ def run(api_port=8082, address=None, scheduler=None, responder=None):
     pruner = tornado.ioloop.PeriodicCallback(scheduler.prune, 60000)
     pruner.start()
 
-    def shutdown_handler(foo=None, bar=None):
+    def shutdown_handler(sig_num=None, trace=None):
         logger.info("Scheduler instance shutting down")
         scheduler.dump()
-        os._exit(0)
+        stop()
+        sys.exit(0)
```

Comment:
I'm sure if there was a reason to use `os._exit(0)` instead of the more gently `sys.exit(0)`, but it prevents coverage to write coverage file.